docs/source/java/install.rst

```diff
@@ -28,7 +28,13 @@ Java modules are regularly built and tested on macOS and Linux distributions.
 Java Compatibility
 ------------------
 
-Java modules are currently compatible with JDK 8, 9, 10, 11, 17, and 18, but only JDK 11 is tested in CI.
+Java modules are currently compatible with JDK 8, 9, 10, 11, 17, and 18.
+Currently, JDK 8, 11, 17, and 18 are tested in CI.
+
+When using Java 9 or later, some JDK internals must be exposed by
```

Comment:
I agree with this note to be more generic.

But only to maintain as a documentation: The main problem was with JDK17 by `it will no longer be possible to relax the strong encapsulation of internal elements using a single command-line option, as it was possible in JDK 9 through JDK 16.` and the jvm argument `--add-opens=java.base/java.nio=ALL-UNNAMED` will need to be used to handle the strong encapsulation violation.

- https://docs.oracle.com/en/java/javase/17/migrate/significant-changes-jdk-release.html#GUID-339B2415-8BA8-438C-93AF-F9C746F7CE45
- https://openjdk.org/jeps/403
- https://bugs.openjdk.org/browse/JDK-8263547

Reply:
So `--illegal-access` no longer works in Java 17. But the `--add-opens` flag works in all these versions, so there's no concern, right?

Reply:
That is right.

Reply:
Thanks. Let's keep it as is then, there's no need to mention a deprecated flag when the more specific one will do.